lib/db/api.js

```diff
@@ -543,25 +543,14 @@ export const lockDeviceByOrigin = function(groups, serial) {
 export const addOriginGroupDevice = function(group, serial) {
     return trace('addOriginGroupDevice', {group, serial}, () => {
         return db.connect().then(client => {
-            return client.collection('groups').findOne({id: group.id}).then((groupData) => {
-                let devices = groupData.devices
-                if (!devices.includes(serial)) {
-                    devices.push(serial)
-                }
-                else {
-                    return getGroup(group.id)
-                }
-                return client.collection('groups').updateOne(
-                    {
-                        id: group.id
-                    },
-                    {
-                        $set: {
-                            devices: devices
-                        }
-                    }).then(function() {
-                    return getGroup(group.id)
-                })
+            if (group.devices.includes(serial)) {
+                return getGroup(group.id)
+            }
```

Comment:
Changing $push to $addToSet will make this check obsolete and in a way make the funciton more atomic